Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -1206,29 +1205,40 @@ class SwiftObjectServerTests: SwiftSyncTestCase {
             return
 
         }
-        let ex = expectation(description: "streaming-downloads-expectation")
         var hasBeenFulfilled = false
 
+        var ex = expectation(description: "download one")
         let token = session.addProgressNotification(for: .download, mode: .reportIndefinitely) { p in
-            callCount += 1
-            XCTAssertGreaterThanOrEqual(p.transferredBytes, transferred)
-            XCTAssertGreaterThanOrEqual(p.transferrableBytes, transferrable)
-            transferred = p.transferredBytes
-            transferrable = p.transferrableBytes
-            if p.transferredBytes > 0 && p.isTransferComplete && !hasBeenFulfilled {
-                ex.fulfill()
-                hasBeenFulfilled = true
+            DispatchQueue.main.async {
```

Comment:
I think it was a race condition at the end of the test? Without this we can return from the test function while the sync thread is still in the progress notification callback, and maybe that was an issue?

I clearly should have written a comment saying why because I don't remember and there isn't an obvious reason. Might just be incorrect.